server/agent/src/main/java/ai/starwhale/mlops/agent/taskexecutor/SourcePool.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022.1-2022
+ * StarWhale.ai All right reserved. This software is the confidential and proprietary information of
+ * StarWhale.ai ("Confidential Information"). You shall not disclose such Confidential Information and shall use it only
+ * in accordance with the terms of the license agreement you entered into with StarWhale.com.
+ */
+
+package ai.starwhale.mlops.agent.taskexecutor;
+
+import ai.starwhale.mlops.domain.node.Device;
+import java.util.ArrayDeque;
+import java.util.HashSet;
+import java.util.Queue;
+import java.util.Set;
+import javax.annotation.PostConstruct;
+
+public class SourcePool {
+
+    /**
+     * ready to work
+     */
+    private static volatile boolean ready = false;
```

Comment:
where is ready set to true

Reply:
this class have an init method，but now is todo state